### PR TITLE
2966/unauthorized agents shown

### PR DESCRIPTION
**Pull request by sortizowlh**
Hi team!
This PR resolve both issues:
#2966 and #2986

### CODE REVIEW

**Comment by CPAlejandro** [comment]
I am checking the PR and I have found the following error:
I have created a user with the policies that I show and when entering the / agents section of my user, it appears as the most active user, one to which I do not have permissionsimage
image

**Comment by Desvelao** [comment]
I tested the PR, and it seems it working for me.

Screenshots:
image
image
image
image


Could someone test it?

I am not sure what happened, but, why are there so many commits and files modified?

File: public/controllers/agent/agents-preview.js [hunk 1]
```diff
@@ -194,20 +197,23 @@ export class AgentsPreviewController {
...
+
       if (info.data.data === '' && this.mostActiveAgent.name !== '') {
         this.mostActiveAgent.id = '000';
       } else {
         this.mostActiveAgent.id = info.data.data;
       }
       return this.mostActiveAgent;
-    } catch (error) { }
+    } catch (error) { 
+    }
```
Comment (gabiwassan) [inline public/controllers/agent/agents-preview.js:216]: This catch is not doing anything, can we add a toast to show the possible error?

File: public/kibana-integrations/kibana-vis.js [hunk 2]
```diff
@@ -205,14 +246,21 @@ class KibanaVis extends Component {
...
         isAgentStatus && timeFilterSeconds < 900
           ? { from: "now-15m", to: "now", mode: "quick" }
           : timefilter.getTime();
       const filters = isAgentStatus ? [] : discoverList[1] || [];
       const query = !isAgentStatus ? discoverList[0] : {};
-
+      const rawVis = raw ? raw.filter((item) => item && item.id === this.visID) : [];
+      let vizPattern;
+      try {
+        vizPattern = JSON.parse(rawVis[0].attributes.kibanaSavedObjectMeta.searchSourceJSON).index;
```
Comment (gabiwassan) [inline public/kibana-integrations/kibana-vis.js:260]: Same here, in this case, we have hidden the errors. Maybe we can log this kind of error, it's not necessary to show a toast.

**Review by gabiwassan — CHANGES_REQUESTED** [review]
Just some comments! Great Job 👍🏼

**Comment by frankeros** [comment]
moved to #3088

**Pull request closed without merging**